src/backend/distributed/executor/multi_utility.c

```diff
@@ -1093,23 +1095,20 @@ ExecuteDistributedDDLCommand(Oid relationId, const char *ddlCommandString,
 
 
 /*
- * SwitchTo2PCForTransaction sets the citus.multi_shard_commit_protocol value
- * to '2pc' for the current transaction, in order to force 2PC even the value
- * of citus.multi_shard_commit_protocol is '1pc';
+ * ShowNoticeIfNotUsing2PC shows a notice message about using 2PC by setting
+ * citus.multi_shard_commit_protocol to 2PC. The notice message is shown only once in a
+ * session
  */
 static void
-SetLocalCommitProtocolTo2PC(void)
+ShowNoticeIfNotUsing2PC(void)
 {
-	if (MultiShardCommitProtocol != COMMIT_PROTOCOL_2PC)
+	if (MultiShardCommitProtocol != COMMIT_PROTOCOL_2PC && !warnedUserAbout2PC)
 	{
-		ereport(DEBUG2, (errmsg("switching to 2PC for the transaction")));
-
-		set_config_option("citus.multi_shard_commit_protocol",
-						  "2pc",
-						  PGC_USERSET,
-						  PGC_S_SESSION,
-						  GUC_ACTION_LOCAL,
-						  true, 0, false);
+		ereport(NOTICE, (errmsg("using one-phase commit for distributed DDL commands"),
+						 errhint("You can SET citus.multi_shard_commit_protocol to '2pc' "
```

Comment:
I wonder whether we can communicate a bit more clearly that the SET... is actually a command you can run prior to your DDL, e.g.:

    To use two-phase commit for extra safety, you can use: SET citus.multi_shard_commit_protocol TO '2pc'